lib/restforce/mash.rb

```diff
@@ -28,6 +28,8 @@ def klass(val)
           # of sobject records.
           Restforce::Collection
         elsif val.key? 'attributes'
+          return Restforce::SObject unless val['attributes']
+
           case (val['attributes']['type'])
```

Comment:
@apurkiss just a suggestion: i think you can change this line to `val.dig('attributes', 'type')` and remove the guard clause, which would be slightly less complex maybe?